src/core/mount.c

```diff
@@ -1622,6 +1622,12 @@ static int mount_setup_existing_unit(
 
         MOUNT(u)->from_proc_self_mountinfo = true;
 
+        /* when we have both info from /etc/fstab and from /proc/self/mountinfo, and the mount generated by
+         * fstab is in a failed state then remove any BindsTo= deps generated due to fstab, and instead
+         * snthesize BindsTo= on the block device listed in /proc/self/mountinfo. */
```

Comment:
This comment has a bunch of typos: when → When, snthesize. Mounts are not "generated by fstab", at most "generated **from** fstab".